File: WEEK 1/Python/Profit Calculation for a Store.py

def calculate_prices(original_price, markup_percentage, sales_tax_rate):

    markup_amount = (markup_percentage / 100) * original_price

    selling_price = original_price + markup_amount

    sales_tax = (sales_tax_rate / 100) * selling_price

    final_price = selling_price + sales_tax

    return selling_price, sales_tax, final_price

# Main program
try:
    # Ask the user for the original price
    original_price = float(input("Enter the original price of the item (in R): "))

    # Ask the user for the markup percentage
    markup_percentage = float(input("Enter the percentage of the mark-up: "))

    # Ask the user for the sales tax rate
    sales_tax_rate = float(input("Enter the sales tax rate (in percentage): "))

    # Calculate prices using the function
    selling_price, sales_tax, final_price = calculate_prices(original_price, markup_percentage, sales_tax_rate)

    # Display the results in South African English
    print(f"Original price of the item: R {original_price:.2f}")
    print(f"Percentage of the mark-up: {markup_percentage:.2f}%")
    print(f"Store’s selling price of the item: R {selling_price:.2f}")
    print(f"Sales tax rate: {sales_tax_rate:.2f}%")
    print(f"Sales tax: R {sales_tax:.2f}")
    print(f"Final price of the item: R {final_price:.2f}")

except ValueError:
    print("Invalid input. Please enter valid numeric values for the inputs (in R and percentage).")

File: WEEK 1/C++/Profit Calculation for a Store.cpp
#include <iostream>

int main() {
    double originalPrice, markupPercentage, salesTaxRate;

    std::cout << "Enter the original price of the item: $";
    std::cin >> originalPrice;

    std::cout << "Enter the percentage of the mark-up: ";
    std::cin >> markupPercentage;

    std::cout << "Enter the sales tax rate: ";
    std::cin >> salesTaxRate;

    double markupAmount = (markupPercentage / 100) * originalPrice;
    double sellingPrice = originalPrice + markupAmount;
    double salesTax = (salesTaxRate / 100) * sellingPrice;
    double finalPrice = sellingPrice + salesTax;

    std::cout << "Original Price: $" << originalPrice << std::endl;
    std::cout << "Mark-up Percentage: " << markupPercentage << "%" << std::endl;
    std::cout << "Selling Price: $" << sellingPrice << std::endl;
    std::cout << "Sales Tax Rate: " << salesTaxRate << "%" << std::endl;
    std::cout << "Sales Tax: $" << salesTax << std::endl;
    std::cout << "Final Price: $" << finalPrice << std::endl;

    return 0;
}
